File: 03-frontend/react-movies/src/models/AddMovieRequest.ts
class AddMovieRequest {
    title: string;
    img: string;
    trailer: string;
    iframe: string;
    description: string;
    releaseDate: Date;
    genre: string;
    rating: number;
    director: string;
    cast: string;
    writers: string;
    countries: string;
    languages: string;
    runtime: number; 

    constructor (
        title: string,
        img: string,
        trailer: string,
        iframe: string,
        description: string,
        releaseDate: Date,
        genre: string,
        rating: number,
        director: string,
        cast: string,
        writers: string,
        countries: string,
        languages: string,
        runtime: number
    ) {
        this.title = title;
        this.img = img;
        this.trailer = trailer;
        this.iframe = iframe;
        this.description = description;
        this.releaseDate = releaseDate;
        this.genre = genre;
        this.rating = rating;
        this.director = director;
        this.cast = cast;
        this.writers = writers;
        this.countries = countries;
        this.languages = languages;
        this.runtime = runtime;
    }
}

export default AddMovieRequest;

File: 02-backend/spring-boot-movies/src/main/java/com/kcakmak/spring_boot_movies/requestmodels/AddMovieRequest.java
package com.kcakmak.spring_boot_movies.requestmodels;

import java.math.BigDecimal;
import java.sql.Date;

public class AddMovieRequest {

    private String title;

    private String img;

    private String trailer;

    private String iframe;

    private String description;

    private Date releaseDate;

    private String genre;

    private BigDecimal rating;

    private String director;

    private String cast;

    private String writers;

    private String countries;

    private String languages;

    private Integer runtime;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    public String getTrailer() {
        return trailer;
    }

    public void setTrailer(String trailer) {
        this.trailer = trailer;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getIframe() {
        return iframe;
    }

    public void setIframe(String iframe) {
        this.iframe = iframe;
    }

    public Date getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(Date releaseDate) {
        this.releaseDate = releaseDate;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public BigDecimal getRating() {
        return rating;
    }

    public void setRating(BigDecimal rating) {
        this.rating = rating;
    }

    public String getDirector() {
        return director;
    }

    public void setDirector(String director) {
        this.director = director;
    }

    public String getCast() {
        return cast;
    }

    public void setCast(String cast) {
        this.cast = cast;
    }

    public String getWriters() {
        return writers;
    }

    public void setWriters(String writers) {
        this.writers = writers;
    }

    public String getCountries() {
        return countries;
    }

    public void setCountries(String countries) {
        this.countries = countries;
    }

    public String getLanguages() {
        return languages;
    }

    public void setLanguages(String languages) {
        this.languages = languages;
    }

    public Integer getRuntime() {
        return runtime;
    }

    public void setRuntime(Integer runtime) {
        this.runtime = runtime;
    }
}
